daemon/daemon.go

```diff
@@ -365,33 +365,39 @@ func (daemon *Daemon) restore() error {
 
 				c.RestartManager().Cancel() // manually start containers because some need to wait for swarm networking
 
-				if c.IsPaused() && alive {
-					s, err := daemon.containerd.Status(context.Background(), c.ID)
-					if err != nil {
-						logger(c).WithError(err).Error("failed to get container status")
-					} else {
-						logger(c).WithField("state", s).Info("restored container paused")
-						switch s {
-						case containerd.Paused, containerd.Pausing:
-							// nothing to do
-						case containerd.Stopped:
-							alive = false
-						case containerd.Unknown:
-							log.Error("unknown status for paused container during restore")
-						default:
-							// running
-							c.Lock()
-							c.Paused = false
-							daemon.setStateCounter(c)
-							if err := c.CheckpointTo(daemon.containersReplica); err != nil {
-								log.WithError(err).Error("failed to update paused container state")
+				if alive {
+					if c.IsPaused() {
+						s, err := daemon.containerd.Status(context.Background(), c.ID)
+						if err != nil {
+							logger(c).WithError(err).Error("failed to get container status")
+						} else {
+							logger(c).WithField("state", s).Info("restored container paused")
+							switch s {
+							case containerd.Paused, containerd.Pausing:
+								// nothing to do
+							case containerd.Stopped:
+								alive = false
```

Comment:
Oh! hmm.. wondering now if we need the two separate if blocks after all. The linter now marks this one, which may be a valid issue. Looks like we're updating the `alive` status based on the container status, which  means that we'll have to run the `"setting stopped state"` code further down (which is now in an "else")

```
[2021-09-22T19:57:59.011Z] daemon/daemon.go:379:9: ineffectual assignment to `alive` (ineffassign)
[2021-09-22T19:57:59.011Z] 								alive = false
[2021-09-22T19:57:59.011Z] 								^
```


Sorry for that (and for suggesting to change it to a single `if/else`

We really need to review the code logic in this part, as it's too complicated (and error-prone, which the above shows)

Reply:
Yes indeed I did not realize it but it will not work as expected :(
As you say, the "setting stopped state" part is never executed when we enter the "containerd.Stopped" case.

So I think the best compromise would be to go back to the switch statement? It sounds more maintainable to me than multiple if / else statements.

Reply:
Yes, I think we might as well switch back to what we had (to keep the diff small), and after that have a look at how we can refactor this code to make it cleaner/better (not as part of this PR)